
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://microbiomedata.github.io/nmdc-schema/FileTypeEnum/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.29">
    
    
      
        <title>Enum: FileTypeEnum - NMDC Schema Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.76a95c52.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#enum-filetypeenum" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NMDC Schema Documentation" class="md-header__button md-logo" aria-label="NMDC Schema Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NMDC Schema Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Enum: FileTypeEnum
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/microbiomedata/nmdc-schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NMDC Schema Documentation" class="md-nav__button md-logo" aria-label="NMDC Schema Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NMDC Schema Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/microbiomedata/nmdc-schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    About
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://linkml.io/linkml/howtos/collaborative-development.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    How to run a collaborative data modeling project
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../identifiers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifiers in NMDC
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../prefixes_curies_ids_mappings_etc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifiers, CURIes, Prefixes, etc.
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Metadata_Documentation_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Metadata Standards Documentation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://docs.google.com/presentation/d/1ZH41QAoESUwAkdHyUxlrmSKS5M-bT0TOulBgX4rBx2A/edit#slide=id.g26390794265_0_693" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema Contributors How-to from Schema Hackathon 2023-12-10
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../schema-validation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema Validation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../schema_element_deprecation_guide/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Schema element deprecation guide
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../visualizations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualizations
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#permissible-values" class="md-nav__link">
    <span class="md-ellipsis">
      Permissible Values
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#identifier-and-mapping-information" class="md-nav__link">
    <span class="md-ellipsis">
      Identifier and Mapping Information
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Identifier and Mapping Information">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#schema-source" class="md-nav__link">
    <span class="md-ellipsis">
      Schema Source
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#linkml-source" class="md-nav__link">
    <span class="md-ellipsis">
      LinkML Source
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="enum-filetypeenum">Enum: FileTypeEnum</h1>
<p>URI: <a href="./">FileTypeEnum</a></p>
<h2 id="permissible-values">Permissible Values</h2>
<table>
<thead>
<tr>
<th>Value</th>
<th>Meaning</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Reference Calibration File</td>
<td>None</td>
<td>A file that contains data used to calibrate a natural organic matter or metab...</td>
</tr>
<tr>
<td>Metagenome Raw Reads</td>
<td>None</td>
<td>Interleaved paired-end raw sequencing data</td>
</tr>
<tr>
<td>Metagenome Raw Read 1</td>
<td>None</td>
<td>Read 1 raw sequencing data, aka forward reads</td>
</tr>
<tr>
<td>Metagenome Raw Read 2</td>
<td>None</td>
<td>Read 2 raw sequencing data, aka reverse reads</td>
</tr>
<tr>
<td>FT ICR-MS Analysis Results</td>
<td>None</td>
<td>FT ICR-MS-based molecular formula assignment results table</td>
</tr>
<tr>
<td>GC-MS Metabolomics Results</td>
<td>None</td>
<td>GC-MS-based metabolite assignment results table</td>
</tr>
<tr>
<td>Metaproteomics Workflow Statistics</td>
<td>None</td>
<td>Aggregate workflow statistics file</td>
</tr>
<tr>
<td>Protein Report</td>
<td>None</td>
<td>Filtered protein report file</td>
</tr>
<tr>
<td>Peptide Report</td>
<td>None</td>
<td>Filtered peptide report file</td>
</tr>
<tr>
<td>Unfiltered Metaproteomics Results</td>
<td>None</td>
<td>MSGFjobs and MASIC output file</td>
</tr>
<tr>
<td>Read Count and RPKM</td>
<td>None</td>
<td>Annotation read count and RPKM per feature JSON</td>
</tr>
<tr>
<td>QC non-rRNA R2</td>
<td>None</td>
<td>QC removed rRNA reads (R2) fastq</td>
</tr>
<tr>
<td>QC non-rRNA R1</td>
<td>None</td>
<td>QC removed rRNA reads (R1) fastq</td>
</tr>
<tr>
<td>Metagenome Bins</td>
<td>None</td>
<td>Metagenome bin contigs fasta</td>
</tr>
<tr>
<td>Metagenome HQMQ Bins Compression File</td>
<td>None</td>
<td>Compressed file containing high qulaity and medium quality metagenome bins an...</td>
</tr>
<tr>
<td>Metagenome LQ Bins Compression File</td>
<td>None</td>
<td>Compressed file containing low quality metagenome bins and associated files</td>
</tr>
<tr>
<td>Metagenome Bins Info File</td>
<td>None</td>
<td>File containing version information on the binning workflow</td>
</tr>
<tr>
<td>CheckM Statistics</td>
<td>None</td>
<td>CheckM statistics report</td>
</tr>
<tr>
<td>Metagenome Bins Heatmap</td>
<td>None</td>
<td>The Heatmap presents the pdf file containing the KO analysis results for meta...</td>
</tr>
<tr>
<td>Metagenome Bins Barplot</td>
<td>None</td>
<td>The Bar chart presents the pdf file containing the KO analysis results for me...</td>
</tr>
<tr>
<td>Metagenome Bins Krona Plot</td>
<td>None</td>
<td>The Krona plot presents the HTML file containing the KO analysis results for ...</td>
</tr>
<tr>
<td>Read Based Analysis Info File</td>
<td>None</td>
<td>File containing reads based analysis information</td>
</tr>
<tr>
<td>GTDBTK Bacterial Summary</td>
<td>None</td>
<td>GTDBTK bacterial summary</td>
</tr>
<tr>
<td>GTDBTK Archaeal Summary</td>
<td>None</td>
<td>GTDBTK archaeal summary</td>
</tr>
<tr>
<td>GOTTCHA2 Krona Plot</td>
<td>None</td>
<td>GOTTCHA2 krona plot HTML file</td>
</tr>
<tr>
<td>GOTTCHA2 Classification Report</td>
<td>None</td>
<td>GOTTCHA2 classification report file</td>
</tr>
<tr>
<td>GOTTCHA2 Report Full</td>
<td>None</td>
<td>GOTTCHA2 report file</td>
</tr>
<tr>
<td>Kraken2 Krona Plot</td>
<td>None</td>
<td>Kraken2 krona plot HTML file</td>
</tr>
<tr>
<td>Centrifuge Krona Plot</td>
<td>None</td>
<td>Centrifuge krona plot HTML file</td>
</tr>
<tr>
<td>Centrifuge output report file</td>
<td>None</td>
<td>Centrifuge output report file</td>
</tr>
<tr>
<td>Kraken2 Classification Report</td>
<td>None</td>
<td>Kraken2 output report file</td>
</tr>
<tr>
<td>Kraken2 Taxonomic Classification</td>
<td>None</td>
<td>Kraken2 output read classification file</td>
</tr>
<tr>
<td>Centrifuge Classification Report</td>
<td>None</td>
<td>Centrifuge output report file</td>
</tr>
<tr>
<td>Centrifuge Taxonomic Classification</td>
<td>None</td>
<td>Centrifuge output read classification file</td>
</tr>
<tr>
<td>Structural Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with structural annotations</td>
</tr>
<tr>
<td>Structural Annotation Stats Json</td>
<td>None</td>
<td>Structural annotations stats json</td>
</tr>
<tr>
<td>Functional Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with functional annotations</td>
</tr>
<tr>
<td>Annotation Info File</td>
<td>None</td>
<td>File containing annotation info</td>
</tr>
<tr>
<td>Annotation Amino Acid FASTA</td>
<td>None</td>
<td>FASTA amino acid file for annotated proteins</td>
</tr>
<tr>
<td>Annotation Enzyme Commission</td>
<td>None</td>
<td>Tab delimited file for EC annotation</td>
</tr>
<tr>
<td>Annotation KEGG Orthology</td>
<td>None</td>
<td>Tab delimited file for KO annotation</td>
</tr>
<tr>
<td>Assembly Info File</td>
<td>None</td>
<td>File containing assembly info</td>
</tr>
<tr>
<td>Assembly Coverage BAM</td>
<td>None</td>
<td>Sorted bam file of reads mapping back to the final assembly</td>
</tr>
<tr>
<td>Assembly AGP</td>
<td>None</td>
<td>An AGP format file that describes the assembly</td>
</tr>
<tr>
<td>Assembly Scaffolds</td>
<td>None</td>
<td>Final assembly scaffolds fasta</td>
</tr>
<tr>
<td>Assembly Contigs</td>
<td>None</td>
<td>Final assembly contigs fasta</td>
</tr>
<tr>
<td>Assembly Coverage Stats</td>
<td>None</td>
<td>Assembled contigs coverage information</td>
</tr>
<tr>
<td>Contig Mapping File</td>
<td>None</td>
<td>Contig mappings between contigs and scaffolds</td>
</tr>
<tr>
<td>Error Corrected Reads</td>
<td>None</td>
<td>Error corrected reads fastq</td>
</tr>
<tr>
<td>Filtered Sequencing Reads</td>
<td>None</td>
<td>Reads QC result fastq (clean data)</td>
</tr>
<tr>
<td>Read Filtering Info File</td>
<td>None</td>
<td>File containing read filtering information</td>
</tr>
<tr>
<td>QC Statistics Extended</td>
<td>None</td>
<td>Extended report including methods and results for read filtering</td>
</tr>
<tr>
<td>QC Statistics</td>
<td>None</td>
<td>Reads QC summary statistics</td>
</tr>
<tr>
<td>TIGRFam Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with TIGRfam</td>
</tr>
<tr>
<td>CRT Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with CRT</td>
</tr>
<tr>
<td>Genemark Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with Genemark</td>
</tr>
<tr>
<td>Prodigal Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with Prodigal</td>
</tr>
<tr>
<td>TRNA Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with TRNA</td>
</tr>
<tr>
<td>Misc Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with Misc</td>
</tr>
<tr>
<td>RFAM Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with RFAM</td>
</tr>
<tr>
<td>TMRNA Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with TMRNA</td>
</tr>
<tr>
<td>Crispr Terms</td>
<td>None</td>
<td>Crispr Terms</td>
</tr>
<tr>
<td>Product Names</td>
<td>None</td>
<td>Product names file</td>
</tr>
<tr>
<td>Gene Phylogeny tsv</td>
<td>None</td>
<td>Gene Phylogeny tsv</td>
</tr>
<tr>
<td>Scaffold Lineage tsv</td>
<td>None</td>
<td>phylogeny at the scaffold level</td>
</tr>
<tr>
<td>Clusters of Orthologous Groups (COG) Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with COGs</td>
</tr>
<tr>
<td>KO_EC Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with KO_EC</td>
</tr>
<tr>
<td>CATH FunFams (Functional Families) Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with CATH FunFams</td>
</tr>
<tr>
<td>SUPERFam Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with SUPERFam</td>
</tr>
<tr>
<td>SMART Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with SMART</td>
</tr>
<tr>
<td>Pfam Annotation GFF</td>
<td>None</td>
<td>GFF3 format file with Pfam</td>
</tr>
<tr>
<td>Annotation Statistics</td>
<td>None</td>
<td>Annotation statistics report</td>
</tr>
<tr>
<td>Direct Infusion FT ICR-MS Raw Data</td>
<td>None</td>
<td>Direct infusion 21 Tesla Fourier Transform ion cyclotron resonance mass spect...</td>
</tr>
<tr>
<td>LC-DDA-MS/MS Raw Data</td>
<td>None</td>
<td>Liquid chromatographically separated MS1 and Data-Dependent MS2 binary instru...</td>
</tr>
<tr>
<td>Configuration toml</td>
<td>None</td>
<td>A configuration toml file used by various programs to store settings that are...</td>
</tr>
<tr>
<td>LC-MS Lipidomics Results</td>
<td>None</td>
<td>LC-MS-based lipid assignment results table</td>
</tr>
<tr>
<td>Contaminants Amino Acid FASTA</td>
<td>None</td>
<td>FASTA amino acid file for contaminant proteins commonly observed in proteomic...</td>
</tr>
<tr>
<td>Analysis Tool Parameter File</td>
<td>None</td>
<td>A configuration file used by a single computational software tool that stores...</td>
</tr>
<tr>
<td>Workflow Operation Summary</td>
<td>None</td>
<td>A human readable record of analysis steps applied during an instance of a wor...</td>
</tr>
<tr>
<td>Metatranscriptome Expression</td>
<td>None</td>
<td>Metatranscriptome expression values and read counts for gene features predict...</td>
</tr>
<tr>
<td>Metatranscriptome Expression Intergenic</td>
<td>None</td>
<td>Metatranscriptome expression values and read counts for intergenic regions</td>
</tr>
<tr>
<td>Metatranscriptome Expression Info File</td>
<td>None</td>
<td>File containing version information on the expression workflow</td>
</tr>
<tr>
<td>rRNA Filtered Sequencing Reads</td>
<td>None</td>
<td>File containing ribosomal reads from the read qc filtering step</td>
</tr>
<tr>
<td>BAI File</td>
<td>None</td>
<td>An index file found in the same directory as the binary alignment map (BAM) f...</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../data_object_type/">data_object_type</a></td>
<td>The type of file represented by the data object</td>
</tr>
</tbody>
</table>
<h2 id="identifier-and-mapping-information">Identifier and Mapping Information</h2>
<h3 id="schema-source">Schema Source</h3>
<ul>
<li>from schema: <a href="https://w3id.org/nmdc/nmdc">https://w3id.org/nmdc/nmdc</a></li>
</ul>
<h2 id="linkml-source">LinkML Source</h2>
<details>

<pre><code class="language-yaml">name: FileTypeEnum
from_schema: https://w3id.org/nmdc/nmdc
rank: 1000
permissible_values:
  Reference Calibration File:
    text: Reference Calibration File
    description: A file that contains data used to calibrate a natural organic matter
      or metabalomics analysis.
  Metagenome Raw Reads:
    text: Metagenome Raw Reads
    description: Interleaved paired-end raw sequencing data
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: ^\.fastq(\.gz)?$
  Metagenome Raw Read 1:
    text: Metagenome Raw Read 1
    description: Read 1 raw sequencing data, aka forward reads
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: ^.+_R1\.fastq(\.gz)?$
    examples:
    - value: BMI_H25VYBGXH_19S_31WellG1_R1.fastq.gz
  Metagenome Raw Read 2:
    text: Metagenome Raw Read 2
    description: Read 2 raw sequencing data, aka reverse reads
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: ^.+_R2\.fastq(\.gz)?$
    examples:
    - value: BMI_H25VYBGXH_19S_31WellG1_R2.fastq.gz
  FT ICR-MS Analysis Results:
    text: FT ICR-MS Analysis Results
    description: FT ICR-MS-based molecular formula assignment results table
  GC-MS Metabolomics Results:
    text: GC-MS Metabolomics Results
    description: GC-MS-based metabolite assignment results table
  Metaproteomics Workflow Statistics:
    text: Metaproteomics Workflow Statistics
    description: Aggregate workflow statistics file
  Protein Report:
    text: Protein Report
    description: Filtered protein report file
  Peptide Report:
    text: Peptide Report
    description: Filtered peptide report file
  Unfiltered Metaproteomics Results:
    text: Unfiltered Metaproteomics Results
    description: MSGFjobs and MASIC output file
  Read Count and RPKM:
    text: Read Count and RPKM
    description: Annotation read count and RPKM per feature JSON
  QC non-rRNA R2:
    text: QC non-rRNA R2
    description: QC removed rRNA reads (R2) fastq
  QC non-rRNA R1:
    text: QC non-rRNA R1
    description: QC removed rRNA reads (R1) fastq
  Metagenome Bins:
    text: Metagenome Bins
    description: Metagenome bin contigs fasta
  Metagenome HQMQ Bins Compression File:
    text: Metagenome HQMQ Bins Compression File
    description: Compressed file containing high qulaity and medium quality metagenome
      bins and associated files
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_hqmq_bin.zip'
  Metagenome LQ Bins Compression File:
    text: Metagenome LQ Bins Compression File
    description: Compressed file containing low quality metagenome bins and associated
      files
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_lq_bin.zip'
  Metagenome Bins Info File:
    text: Metagenome Bins Info File
    description: File containing version information on the binning workflow
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_bin.info'
  CheckM Statistics:
    text: CheckM Statistics
    description: CheckM statistics report
  Metagenome Bins Heatmap:
    text: Metagenome Bins Heatmap
    description: The Heatmap presents the pdf file containing the KO analysis results
      for metagenome bins
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_heatmap.pdf'
  Metagenome Bins Barplot:
    text: Metagenome Bins Barplot
    description: The Bar chart presents the pdf file containing the KO analysis results
      for metagenome bins
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_barplot.pdf'
  Metagenome Bins Krona Plot:
    text: Metagenome Bins Krona Plot
    description: The Krona plot presents the HTML file containing the KO analysis
      results for metagenome bins
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[mag_wf_activity_id]_kronaplot.html'
  Read Based Analysis Info File:
    text: Read Based Analysis Info File
    description: File containing reads based analysis information
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: profiler.info
  GTDBTK Bacterial Summary:
    text: GTDBTK Bacterial Summary
    description: GTDBTK bacterial summary
  GTDBTK Archaeal Summary:
    text: GTDBTK Archaeal Summary
    description: GTDBTK archaeal summary
  GOTTCHA2 Krona Plot:
    text: GOTTCHA2 Krona Plot
    description: GOTTCHA2 krona plot HTML file
  GOTTCHA2 Classification Report:
    text: GOTTCHA2 Classification Report
    description: GOTTCHA2 classification report file
  GOTTCHA2 Report Full:
    text: GOTTCHA2 Report Full
    description: GOTTCHA2 report file
  Kraken2 Krona Plot:
    text: Kraken2 Krona Plot
    description: Kraken2 krona plot HTML file
  Centrifuge Krona Plot:
    text: Centrifuge Krona Plot
    description: Centrifuge krona plot HTML file
  Centrifuge output report file:
    text: Centrifuge output report file
    description: Centrifuge output report file
  Kraken2 Classification Report:
    text: Kraken2 Classification Report
    description: Kraken2 output report file
  Kraken2 Taxonomic Classification:
    text: Kraken2 Taxonomic Classification
    description: Kraken2 output read classification file
  Centrifuge Classification Report:
    text: Centrifuge Classification Report
    description: Centrifuge output report file
  Centrifuge Taxonomic Classification:
    text: Centrifuge Taxonomic Classification
    description: Centrifuge output read classification file
  Structural Annotation GFF:
    text: Structural Annotation GFF
    description: GFF3 format file with structural annotations
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_structural_annotation.gff'
  Structural Annotation Stats Json:
    text: Structural Annotation Stats Json
    description: Structural annotations stats json
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_structural_annotation_stats.json'
  Functional Annotation GFF:
    text: Functional Annotation GFF
    description: GFF3 format file with functional annotations
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_functional_annotation.gff'
  Annotation Info File:
    text: Annotation Info File
    description: File containing annotation info
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_imgap.info'
  Annotation Amino Acid FASTA:
    text: Annotation Amino Acid FASTA
    description: FASTA amino acid file for annotated proteins
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_proteins.faa'
  Annotation Enzyme Commission:
    text: Annotation Enzyme Commission
    description: Tab delimited file for EC annotation
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_ec.tsv'
  Annotation KEGG Orthology:
    text: Annotation KEGG Orthology
    description: Tab delimited file for KO annotation
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_ko.tsv'
  Assembly Info File:
    text: Assembly Info File
    description: File containing assembly info
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: README.txt
  Assembly Coverage BAM:
    text: Assembly Coverage BAM
    description: Sorted bam file of reads mapping back to the final assembly
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_pairedMapped.sam.gz'
  Assembly AGP:
    text: Assembly AGP
    description: An AGP format file that describes the assembly
  Assembly Scaffolds:
    text: Assembly Scaffolds
    description: Final assembly scaffolds fasta
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_assembly.contigs.fasta'
  Assembly Contigs:
    text: Assembly Contigs
    description: Final assembly contigs fasta
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: assembly.contigs.fasta
  Assembly Coverage Stats:
    text: Assembly Coverage Stats
    description: Assembled contigs coverage information
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_pairedMapped_sorted.bam.cov'
  Contig Mapping File:
    text: Contig Mapping File
    description: Contig mappings between contigs and scaffolds
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_contig_names_mapping.tsv'
  Error Corrected Reads:
    text: Error Corrected Reads
    description: Error corrected reads fastq
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: input.corr.fastq.gz
  Filtered Sequencing Reads:
    text: Filtered Sequencing Reads
    description: Reads QC result fastq (clean data)
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '/.+?(?=filter)/filter-METAGENOME.fastq.gz '
  Read Filtering Info File:
    text: Read Filtering Info File
    description: File containing read filtering information
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[rqc_wf_activity_id]_readsQC.info'
  QC Statistics Extended:
    text: QC Statistics Extended
    description: Extended report including methods and results for read filtering
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: /.+?(?=filter)/filtered-report.txt
  QC Statistics:
    text: QC Statistics
    description: Reads QC summary statistics
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[rqc_wf_activity_id]_filterStats2.txt'
  TIGRFam Annotation GFF:
    text: TIGRFam Annotation GFF
    description: GFF3 format file with TIGRfam
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_tigrfam.gff'
  CRT Annotation GFF:
    text: CRT Annotation GFF
    description: GFF3 format file with CRT
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_crt.gff'
  Genemark Annotation GFF:
    text: Genemark Annotation GFF
    description: GFF3 format file with Genemark
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_genemark.gff'
  Prodigal Annotation GFF:
    text: Prodigal Annotation GFF
    description: GFF3 format file with Prodigal
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_prodigal.gff'
  TRNA Annotation GFF:
    text: TRNA Annotation GFF
    description: GFF3 format file with TRNA
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_trna.gff'
  Misc Annotation GFF:
    text: Misc Annotation GFF
    description: GFF3 format file with Misc
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_rfam_misc_bind_misc_feature_regulatory.gff'
  RFAM Annotation GFF:
    text: RFAM Annotation GFF
    description: GFF3 format file with RFAM
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_rfam.gff'
  TMRNA Annotation GFF:
    text: TMRNA Annotation GFF
    description: GFF3 format file with TMRNA
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_rfam_ncrna_tmrna.gff'
  Crispr Terms:
    text: Crispr Terms
    description: Crispr Terms
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_crt.crisprs'
  Product Names:
    text: Product Names
    description: Product names file
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_product_names.tsv'
  Gene Phylogeny tsv:
    text: Gene Phylogeny tsv
    description: Gene Phylogeny tsv
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_gene_phylogeny.tsv'
  Scaffold Lineage tsv:
    text: Scaffold Lineage tsv
    description: phylogeny at the scaffold level
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_scaffold_lineage.tsv'
  Clusters of Orthologous Groups (COG) Annotation GFF:
    text: Clusters of Orthologous Groups (COG) Annotation GFF
    description: GFF3 format file with COGs
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_cog.gff'
  KO_EC Annotation GFF:
    text: KO_EC Annotation GFF
    description: GFF3 format file with KO_EC
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_ko_ec.gff'
  CATH FunFams (Functional Families) Annotation GFF:
    text: CATH FunFams (Functional Families) Annotation GFF
    description: GFF3 format file with CATH FunFams
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_cath_funfam.gff'
  SUPERFam Annotation GFF:
    text: SUPERFam Annotation GFF
    description: GFF3 format file with SUPERFam
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_supfam.gff'
  SMART Annotation GFF:
    text: SMART Annotation GFF
    description: GFF3 format file with SMART
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_smart.gff'
  Pfam Annotation GFF:
    text: Pfam Annotation GFF
    description: GFF3 format file with Pfam
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '[GOLD-AP]_pfam.gff'
  Annotation Statistics:
    text: Annotation Statistics
    description: Annotation statistics report
  Direct Infusion FT ICR-MS Raw Data:
    text: Direct Infusion FT ICR-MS Raw Data
    description: Direct infusion 21 Tesla Fourier Transform ion cyclotron resonance
      mass spectrometry raw data acquired in broadband full scan mode
  LC-DDA-MS/MS Raw Data:
    text: LC-DDA-MS/MS Raw Data
    description: Liquid chromatographically separated MS1 and Data-Dependent MS2 binary
      instrument file
  Configuration toml:
    text: Configuration toml
    description: A configuration toml file used by various programs to store settings
      that are specific to their respective software.
    broad_mappings:
    - edam.format:4005
  LC-MS Lipidomics Results:
    text: LC-MS Lipidomics Results
    description: LC-MS-based lipid assignment results table.
  Contaminants Amino Acid FASTA:
    text: Contaminants Amino Acid FASTA
    description: FASTA amino acid file for contaminant proteins commonly observed
      in proteomics data.
  Analysis Tool Parameter File:
    text: Analysis Tool Parameter File
    description: A configuration file used by a single computational software tool
      that stores settings that are specific to that tool.
  Workflow Operation Summary:
    text: Workflow Operation Summary
    description: A human readable record of analysis steps applied during an instance
      of a workflow operation.
  Metatranscriptome Expression:
    text: Metatranscriptome Expression
    description: Metatranscriptome expression values and read counts for gene features
      predicted on contigs
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '*.rnaseq_gea.txt'
  Metatranscriptome Expression Intergenic:
    text: Metatranscriptome Expression Intergenic
    description: Metatranscriptome expression values and read counts for intergenic
      regions.
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '*.rnaseq_gea.intergenic.txt'
  Metatranscriptome Expression Info File:
    text: Metatranscriptome Expression Info File
    description: File containing version information on the expression workflow
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '*_readcount.info'
  rRNA Filtered Sequencing Reads:
    text: rRNA Filtered Sequencing Reads
    description: File containing ribosomal reads from the read qc filtering step.
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '*.rRNA.fastq.gz'
  BAI File:
    text: BAI File
    description: An index file found in the same directory as the binary alignment
      map (BAM) file, a compressed binary version of a sequence alignment/map (SAM)
      file.
    annotations:
      file_name_pattern:
        tag: file_name_pattern
        value: '*.pairedMapped_sorted.bam.bai'
    exact_mappings:
    - NCIT:C190163

</code></pre>

</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>