
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://microbiomedata.github.io/nmdc-schema/WorkflowExecution/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.29">
    
    
      
        <title>Class: WorkflowExecution - NMDC Schema Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.76a95c52.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#class-workflowexecution" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NMDC Schema Documentation" class="md-header__button md-logo" aria-label="NMDC Schema Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NMDC Schema Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Class: WorkflowExecution
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/microbiomedata/nmdc-schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NMDC Schema Documentation" class="md-nav__button md-logo" aria-label="NMDC Schema Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NMDC Schema Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/microbiomedata/nmdc-schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    About
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://linkml.io/linkml/howtos/collaborative-development.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    How to run a collaborative data modeling project
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../identifiers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifiers in NMDC
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../prefixes_curies_ids_mappings_etc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifiers, CURIes, Prefixes, etc.
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Metadata_Documentation_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Metadata Standards Documentation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://docs.google.com/presentation/d/1ZH41QAoESUwAkdHyUxlrmSKS5M-bT0TOulBgX4rBx2A/edit#slide=id.g26390794265_0_693" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema Contributors How-to from Schema Hackathon 2023-12-10
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../schema-validation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NMDC Schema Validation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../schema_element_deprecation_guide/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Schema element deprecation guide
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../visualizations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualizations
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#inheritance" class="md-nav__link">
    <span class="md-ellipsis">
      Inheritance
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usages" class="md-nav__link">
    <span class="md-ellipsis">
      Usages
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#aliases" class="md-nav__link">
    <span class="md-ellipsis">
      Aliases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comments" class="md-nav__link">
    <span class="md-ellipsis">
      Comments
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#identifier-and-mapping-information" class="md-nav__link">
    <span class="md-ellipsis">
      Identifier and Mapping Information
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Identifier and Mapping Information">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#schema-source" class="md-nav__link">
    <span class="md-ellipsis">
      Schema Source
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mappings" class="md-nav__link">
    <span class="md-ellipsis">
      Mappings
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#linkml-source" class="md-nav__link">
    <span class="md-ellipsis">
      LinkML Source
    </span>
  </a>
  
    <nav class="md-nav" aria-label="LinkML Source">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#direct" class="md-nav__link">
    <span class="md-ellipsis">
      Direct
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#induced" class="md-nav__link">
    <span class="md-ellipsis">
      Induced
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="class-workflowexecution">Class: WorkflowExecution</h1>
<p><em>Represents an instance of an execution of a particular workflow</em></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This is an abstract class and should not be instantiated directly.</p>
</div>
<p>URI: <a href="https://w3id.org/nmdc/WorkflowExecution">nmdc:WorkflowExecution</a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Mermaid class diagram too large to render.</p>
</div>
<h2 id="inheritance">Inheritance</h2>
<ul>
<li><a href="../NamedThing/">NamedThing</a><ul>
<li><a href="../PlannedProcess/">PlannedProcess</a><ul>
<li><strong>WorkflowExecution</strong><ul>
<li><a href="../MetagenomeAnnotation/">MetagenomeAnnotation</a></li>
<li><a href="../MetagenomeAssembly/">MetagenomeAssembly</a></li>
<li><a href="../MetatranscriptomeAssembly/">MetatranscriptomeAssembly</a></li>
<li><a href="../MetatranscriptomeAnnotation/">MetatranscriptomeAnnotation</a></li>
<li><a href="../MetatranscriptomeExpressionAnalysis/">MetatranscriptomeExpressionAnalysis</a></li>
<li><a href="../MagsAnalysis/">MagsAnalysis</a></li>
<li><a href="../MetagenomeSequencing/">MetagenomeSequencing</a></li>
<li><a href="../ReadQcAnalysis/">ReadQcAnalysis</a></li>
<li><a href="../ReadBasedTaxonomyAnalysis/">ReadBasedTaxonomyAnalysis</a></li>
<li><a href="../MetabolomicsAnalysis/">MetabolomicsAnalysis</a></li>
<li><a href="../MetaproteomicsAnalysis/">MetaproteomicsAnalysis</a></li>
<li><a href="../NomAnalysis/">NomAnalysis</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Cardinality and Range</th>
<th>Description</th>
<th>Inheritance</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../ended_at_time/">ended_at_time</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td></td>
<td>direct</td>
</tr>
<tr>
<td><a href="../execution_resource/">execution_resource</a></td>
<td>1 <br/> <a href="../ExecutionResourceEnum/">ExecutionResourceEnum</a></td>
<td>The computing resource or facility where the workflow was executed</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../git_url/">git_url</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>The url that points to the exact github location of a workflow</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../started_at_time/">started_at_time</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td></td>
<td>direct</td>
</tr>
<tr>
<td><a href="../version/">version</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td></td>
<td>direct</td>
</tr>
<tr>
<td><a href="../was_informed_by/">was_informed_by</a></td>
<td>1 <br/> <a href="../DataGeneration/">DataGeneration</a></td>
<td></td>
<td>direct</td>
</tr>
<tr>
<td><a href="../has_input/">has_input</a></td>
<td>1..* <br/> <a href="../NamedThing/">NamedThing</a></td>
<td>An input to a process</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../has_output/">has_output</a></td>
<td>* <br/> <a href="../NamedThing/">NamedThing</a></td>
<td>An output from a process</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../processing_institution/">processing_institution</a></td>
<td>0..1 <br/> <a href="../ProcessingInstitutionEnum/">ProcessingInstitutionEnum</a></td>
<td>The organization that processed the sample</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../protocol_link/">protocol_link</a></td>
<td>0..1 <br/> <a href="../Protocol/">Protocol</a></td>
<td></td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../start_date/">start_date</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The date on which any process or activity was started</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../end_date/">end_date</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The date on which any process or activity was ended</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../qc_status/">qc_status</a></td>
<td>0..1 <br/> <a href="../StatusEnum/">StatusEnum</a></td>
<td>Stores information about the result of a process (ie the process of sequencin...</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../qc_comment/">qc_comment</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Slot to store additional comments about laboratory or workflow output</td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../has_failure_categorization/">has_failure_categorization</a></td>
<td>* <br/> <a href="../FailureCategorization/">FailureCategorization</a></td>
<td></td>
<td><a href="../PlannedProcess/">PlannedProcess</a></td>
</tr>
<tr>
<td><a href="../id/">id</a></td>
<td>1 <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>A unique identifier for a thing</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../name/">name</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>A human readable label for an entity</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../description/">description</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>a human-readable description of a thing</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../alternative_identifiers/">alternative_identifiers</a></td>
<td>* <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>A list of alternative identifiers for the entity</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../type/">type</a></td>
<td>1 <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>the class_uri of the class that has been instantiated</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
</tbody>
</table>
<h2 id="usages">Usages</h2>
<table>
<thead>
<tr>
<th>used by</th>
<th>used in</th>
<th>type</th>
<th>used</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../FunctionalAnnotationAggMember/">FunctionalAnnotationAggMember</a></td>
<td><a href="../metagenome_annotation_id/">metagenome_annotation_id</a></td>
<td>range</td>
<td><a href="./">WorkflowExecution</a></td>
</tr>
<tr>
<td><a href="../Database/">Database</a></td>
<td><a href="../workflow_execution_set/">workflow_execution_set</a></td>
<td>range</td>
<td><a href="./">WorkflowExecution</a></td>
</tr>
<tr>
<td><a href="../DataObject/">DataObject</a></td>
<td><a href="../was_generated_by/">was_generated_by</a></td>
<td>range</td>
<td><a href="./">WorkflowExecution</a></td>
</tr>
</tbody>
</table>
<h2 id="aliases">Aliases</h2>
<ul>
<li>analysis</li>
</ul>
<h2 id="comments">Comments</h2>
<ul>
<li>Each instance of this (and all other) subclasses of WorkflowExecution is a distinct run with start and stop times, potentially with different inputs and outputs</li>
</ul>
<h2 id="identifier-and-mapping-information">Identifier and Mapping Information</h2>
<h3 id="schema-source">Schema Source</h3>
<ul>
<li>from schema: <a href="https://w3id.org/nmdc/nmdc">https://w3id.org/nmdc/nmdc</a></li>
</ul>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>nmdc:WorkflowExecution</td>
</tr>
<tr>
<td>native</td>
<td>nmdc:WorkflowExecution</td>
</tr>
</tbody>
</table>
<h2 id="linkml-source">LinkML Source</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->

<h3 id="direct">Direct</h3>
<details>

<pre><code class="language-yaml">name: WorkflowExecution
description: Represents an instance of an execution of a particular workflow
alt_descriptions:
  embl.ena:
    source: embl.ena
    description: An analysis contains secondary analysis results derived from sequence
      reads (e.g. a genome assembly)
comments:
- Each instance of this (and all other) subclasses of WorkflowExecution is a distinct
  run with start and stop times, potentially with different inputs and outputs
in_subset:
- workflow subset
from_schema: https://w3id.org/nmdc/nmdc
aliases:
- analysis
is_a: PlannedProcess
abstract: true
slots:
- ended_at_time
- execution_resource
- git_url
- started_at_time
- version
- was_informed_by
slot_usage:
  started_at_time:
    name: started_at_time
    domain_of:
    - WorkflowExecution
    required: true
  git_url:
    name: git_url
    domain_of:
    - WorkflowExecution
    required: true
  has_input:
    name: has_input
    domain_of:
    - PlannedProcess
    required: true
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  has_output:
    name: has_output
    domain_of:
    - PlannedProcess
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  execution_resource:
    name: execution_resource
    domain_of:
    - WorkflowExecution
    required: true
  was_informed_by:
    name: was_informed_by
    domain_of:
    - WorkflowExecution
    required: true
class_uri: nmdc:WorkflowExecution
rules:
- preconditions:
    slot_conditions:
      qc_status:
        name: qc_status
        equals_string: pass
  postconditions:
    slot_conditions:
      has_output:
        name: has_output
        required: true
  description: If qc_status has a value of pass, then the has_output slot is required.
  title: qc_status_pass_has_output_required
- preconditions:
    slot_conditions:
      qc_status:
        name: qc_status
        value_presence: ABSENT
  postconditions:
    slot_conditions:
      has_output:
        name: has_output
        required: true
  description: If qc_status is not specified, then the has_output slot is required.
  title: qc_status_pass_null_has_output_required

</code></pre>

</details>

<h3 id="induced">Induced</h3>
<details>

<pre><code class="language-yaml">name: WorkflowExecution
description: Represents an instance of an execution of a particular workflow
alt_descriptions:
  embl.ena:
    source: embl.ena
    description: An analysis contains secondary analysis results derived from sequence
      reads (e.g. a genome assembly)
comments:
- Each instance of this (and all other) subclasses of WorkflowExecution is a distinct
  run with start and stop times, potentially with different inputs and outputs
in_subset:
- workflow subset
from_schema: https://w3id.org/nmdc/nmdc
aliases:
- analysis
is_a: PlannedProcess
abstract: true
slot_usage:
  started_at_time:
    name: started_at_time
    domain_of:
    - WorkflowExecution
    required: true
  git_url:
    name: git_url
    domain_of:
    - WorkflowExecution
    required: true
  has_input:
    name: has_input
    domain_of:
    - PlannedProcess
    required: true
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  has_output:
    name: has_output
    domain_of:
    - PlannedProcess
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  execution_resource:
    name: execution_resource
    domain_of:
    - WorkflowExecution
    required: true
  was_informed_by:
    name: was_informed_by
    domain_of:
    - WorkflowExecution
    required: true
attributes:
  ended_at_time:
    name: ended_at_time
    notes:
    - 'The regex for ISO-8601 format was taken from here: https://www.myintervals.com/blog/2009/05/20/iso-8601-date-validation-that-doesnt-suck/
      It may not be complete, but it is good enough for now.'
    from_schema: https://w3id.org/nmdc/nmdc
    mappings:
    - prov:endedAtTime
    rank: 1000
    alias: ended_at_time
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: string
    pattern: ^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$
  execution_resource:
    name: execution_resource
    description: The computing resource or facility where the workflow was executed.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: execution_resource
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: ExecutionResourceEnum
    required: true
  git_url:
    name: git_url
    description: The url that points to the exact github location of a workflow.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: git_url
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: string
    required: true
  started_at_time:
    name: started_at_time
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: started_at_time
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: string
    required: true
    pattern: ^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$
  version:
    name: version
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: version
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: string
  was_informed_by:
    name: was_informed_by
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: was_informed_by
    owner: WorkflowExecution
    domain_of:
    - WorkflowExecution
    range: DataGeneration
    required: true
  has_input:
    name: has_input
    description: An input to a process.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: has_input
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: NamedThing
    required: true
    multivalued: true
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  has_output:
    name: has_output
    description: An output from a process.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: has_output
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: NamedThing
    multivalued: true
    structured_pattern:
      syntax: '{id_nmdc_prefix}:(dobj)-{id_shoulder}-{id_blade}$'
      interpolated: true
  processing_institution:
    name: processing_institution
    description: The organization that processed the sample.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: processing_institution
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: ProcessingInstitutionEnum
  protocol_link:
    name: protocol_link
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: protocol_link
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    - Study
    range: Protocol
  start_date:
    name: start_date
    description: The date on which any process or activity was started
    todos:
    - add date string validation pattern
    comments:
    - We are using string representations of dates until all components of our ecosystem
      can handle ISO 8610 dates
    - The date should be formatted as YYYY-MM-DD
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: start_date
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: string
  end_date:
    name: end_date
    description: The date on which any process or activity was ended
    todos:
    - add date string validation pattern
    comments:
    - We are using string representations of dates until all components of our ecosystem
      can handle ISO 8610 dates
    - The date should be formatted as YYYY-MM-DD
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: end_date
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: string
  qc_status:
    name: qc_status
    description: Stores information about the result of a process (ie the process
      of sequencing a library may have for qc_status of 'fail' if not enough data
      was generated)
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: qc_status
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: StatusEnum
  qc_comment:
    name: qc_comment
    description: Slot to store additional comments about laboratory or workflow output.
      For workflow output it may describe the particular workflow stage that failed.
      (ie Failed at call-stage due to a malformed fastq file).
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: qc_comment
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: string
  has_failure_categorization:
    name: has_failure_categorization
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: has_failure_categorization
    owner: WorkflowExecution
    domain_of:
    - PlannedProcess
    range: FailureCategorization
    multivalued: true
    inlined: true
    inlined_as_list: true
  id:
    name: id
    description: A unique identifier for a thing. Must be either a CURIE shorthand
      for a URI or a complete URI
    notes:
    - 'abstracted pattern: prefix:typecode-authshoulder-blade(.version)?(_seqsuffix)?'
    - a minimum length of 3 characters is suggested for typecodes, but 1 or 2 characters
      will be accepted
    - typecodes must correspond 1:1 to a class in the NMDC schema. this will be checked
      via per-class id slot usage assertions
    - minting authority shoulders should probably be enumerated and checked in the
      pattern
    examples:
    - value: nmdc:mgmag-00-x012.1_7_c1
      description: https://github.com/microbiomedata/nmdc-schema/pull/499#discussion_r1018499248
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    identifier: true
    alias: id
    owner: WorkflowExecution
    domain_of:
    - NamedThing
    range: uriorcurie
    required: true
    pattern: ^[a-zA-Z0-9][a-zA-Z0-9_\.]+:[a-zA-Z0-9_][a-zA-Z0-9_\-\/\.,]*$
  name:
    name: name
    description: A human readable label for an entity
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: name
    owner: WorkflowExecution
    domain_of:
    - PersonValue
    - NamedThing
    - Protocol
    range: string
  description:
    name: description
    description: a human-readable description of a thing
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    slot_uri: dcterms:description
    alias: description
    owner: WorkflowExecution
    domain_of:
    - ImageValue
    - NamedThing
    range: string
  alternative_identifiers:
    name: alternative_identifiers
    description: A list of alternative identifiers for the entity.
    from_schema: https://w3id.org/nmdc/nmdc
    rank: 1000
    alias: alternative_identifiers
    owner: WorkflowExecution
    domain_of:
    - MetaboliteIdentification
    - NamedThing
    range: uriorcurie
    multivalued: true
    pattern: ^[a-zA-Z0-9][a-zA-Z0-9_\.]+:[a-zA-Z0-9_][a-zA-Z0-9_\-\/\.,]*$
  type:
    name: type
    description: the class_uri of the class that has been instantiated
    notes:
    - replaces legacy nmdc:type slot
    - makes it easier to read example data files
    - required for polymorphic MongoDB collections
    examples:
    - value: nmdc:Biosample
    - value: nmdc:Study
    from_schema: https://w3id.org/nmdc/nmdc
    see_also:
    - https://github.com/microbiomedata/nmdc-schema/issues/1048
    - https://github.com/microbiomedata/nmdc-schema/issues/1233
    - https://github.com/microbiomedata/nmdc-schema/issues/248
    rank: 1000
    slot_uri: rdf:type
    designates_type: true
    alias: type
    owner: WorkflowExecution
    domain_of:
    - EukEval
    - FunctionalAnnotationAggMember
    - MobilePhaseSegment
    - PortionOfSubstance
    - MagBin
    - MetaboliteIdentification
    - PeptideQuantification
    - ProteinQuantification
    - GenomeFeature
    - FunctionalAnnotation
    - AttributeValue
    - NamedThing
    - FailureCategorization
    - Protocol
    - CreditAssociation
    - Doi
    range: uriorcurie
    required: true
class_uri: nmdc:WorkflowExecution
rules:
- preconditions:
    slot_conditions:
      qc_status:
        name: qc_status
        equals_string: pass
  postconditions:
    slot_conditions:
      has_output:
        name: has_output
        required: true
  description: If qc_status has a value of pass, then the has_output slot is required.
  title: qc_status_pass_has_output_required
- preconditions:
    slot_conditions:
      qc_status:
        name: qc_status
        value_presence: ABSENT
  postconditions:
    slot_conditions:
      has_output:
        name: has_output
        required: true
  description: If qc_status is not specified, then the has_output slot is required.
  title: qc_status_pass_null_has_output_required

</code></pre>

</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>